src/core/mesh/qgsmeshdataset.cpp

```diff
@@ -792,27 +960,50 @@ void QgsMeshMemoryDatasetGroup::clearDatasets()
   memoryDatasets.clear();
 }
 
+void QgsMeshMemoryDatasetGroup::initialize()
+{
+  calculateStatistic();
+}
+
+QString QgsMeshMemoryDatasetGroup::information() const {return QObject::tr( "Memory Dataset Group, do not persit when closing the project" );}
+
 std::shared_ptr<const QgsMeshMemoryDataset> QgsMeshMemoryDatasetGroup::constDataset( int index ) const
 {
   return memoryDatasets[index];
 }
 
-void QgsMeshMemoryDatasetGroup::calculateStatistic()
+QDomElement QgsMeshMemoryDatasetGroup::writeXml( QDomDocument &doc, const QgsReadWriteContext &context ) const
+{
+  Q_UNUSED( doc )
+  Q_UNUSED( context )
+  return QDomElement();
```

Comment:
is this unfinished or this is the real implementations?

Reply:
real implementation. memory dataset groups are not saved so return an invalid QDomElement.

Reply:
best to add in-implementation comment saying that